src/Tasks/Microsoft.NET.Build.Tasks/ReferenceInfo.cs

```diff
@@ -51,10 +51,29 @@ public static IEnumerable<ReferenceInfo> CreateReferenceInfos(IEnumerable<ITaskI
                             r.HasMetadataValue("ReferenceSourceTarget", "ResolveAssemblyReference") &&
                             string.IsNullOrEmpty(r.GetMetadata("NuGetSourceType")));
 
+            return CreateFilteredReferenceInfos(directReferencePaths, referenceSatellitePaths);
+        }
+
+        public static IEnumerable<ReferenceInfo> CreateDependencyReferenceInfos(
+            IEnumerable<ITaskItem> referenceDependencyPaths,
+            IEnumerable<ITaskItem> referenceSatellitePaths)
+        {
+            IEnumerable<ITaskItem> indirectReferencePaths = referenceDependencyPaths
+                .Where(r => r.HasMetadataValue("CopyLocal", "true") &&
+                            string.IsNullOrEmpty(r.GetMetadata("NuGetSourceType")));
```

Comment:
I see NuGetSourceType around elsewhere, but I don't know what it's testing for. I think we may no longer be setting this anywhere.

Reply:
It can probably be removed in a later PR, along with all the other uses, if it isn't actually needed anymore.

Reply:
I did remove it along with a bunch of redundant goop that was filling the log file. Now I don't know how tests are passing, but something has probably regressed. :(

Reply:
There is a regression: #2090 

Unfortunate that we had no test coverage. I will add NuGetSourceType back